samples/bluetooth/llpm/README.rst

```diff
@@ -0,0 +1,313 @@
+
+   * - Parameter
+     - Value
+   * - Connection interval
+     - 80 units (100 ms)
+   * - LLPM connection interval
+     - Lowest interval (1 ms)
+   * - Physical layer (PHY)
+     - LE 2M PHY
+
+
+Requirements
+************
+
+* Two of the following nRF52-series development kit boards:
+
+  * PCA10040
+  * PCA10056
+
+  You can mix different boards.
+* Connection to a computer with a serial terminal for each of the boards.
+
+
+Building and running
+********************
+.. |sample path| replace:: :file:`samples/bluetooth/llpm`
+
+.. include:: /includes/build_and_run.txt
+
+
+Testing
+=======
+
+After programming the sample to both boards, test it by performing the following steps:
+
+1. Connect to both boards with a terminal emulator (for example, PuTTY).
+   See :ref:`putty` for the required settings.
+#. Reset both boards.
+#. Observe that the boards establish a connection.
+   When they are connected, one of them serves as *master* and the other one as *slave*.
+
+   - The master outputs the following information::
+
+       Press any key to set LLPM short connection connection interval (1 ms)
+
+   - The slave outputs the following information::
+
+       Press any key to start measuring transmission latency
+
+#. Press a key in the terminal that is connected to the slave.
+#. Observe that the slave starts sending data to the master and measuring the transmission latency with the default connection interval::
+
+       Transmission Latency: 80917 (us)
+
+#. Press a key in the terminal that is connected to the master.
+#. Observe the output on the master.
+   It is waiting for a user input::
+
+       Press any key to start measuring transmission latency
+
+#. Observe the connection gets updated to LLPM connection interval (1 ms) on both sides::
+
+       Connection intervonal updated: LLPM (1 ms)
+
+#. Observe the output on the slave.
+   While the slave keeps sending data to the master with the LLPM connection interval, the measured latency on the slave becomes approximate 1 ms::
+
+       Transmission Latency: 1098 (us)
+
+#. Press a key in the terminal that is connected to the master.
+#. Observe the output on the master.
+   The master starts sending data to the slave and measuring the transmission latency with LLPM connection interval.
+   The latency remains approximate 1 ms::
+
+       Transmission Latency: 1235 (us)
+
+.. msc::
+   hscale = "1.3";
+   Master,Slave;
+   Master<<=>>Slave         [label="Connected"];
+   Master<<=>>Slave         [label="Discovered BT_UUID_LATENCY SERVICE"];
+   Slave note Slave         [label="Press any key to start measuring transmission latency"];
+   Slave note Slave         [label="Read present timestamp: s1"];
+   Slave=>Master            [label="ATT Write Request (timestamp: s1)"];
+   Master>>Slave            [label="ATT Write response"];
+   Slave note Slave         [label="Read present timestamp: s2"];
+   Slave note Slave         [label="Slave Latency = (s2 - s1) / 2"];
```

Comment:
To avoid confusion, we should rather use the term "Latency, Slave" as "Slave Latency" is a BLE Spec feature. Similar for "Master Latency"

Reply:
To avoid confusion, maybe we should just use "Latency = ...". Because we only divide the RTT in half, it is hard to judge that the latency belongs to master or slave. ??